collections/windowed_test.ts

```diff
@@ -0,0 +1,220 @@
+) {
+  const actual = windowed(...input);
+  assertEquals(actual, expected, message);
+}
+
+Deno.test({
+  name: "[collections/windowed] no mutation",
+  fn() {
+    const numbers = [1, 2, 3, 4, 5];
+    windowed(numbers, 3);
+    assertEquals(numbers, [1, 2, 3, 4, 5]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] empty input",
+  fn() {
+    windowedTest([[], 3], []);
+    windowedTest([[], 3, {}], []);
+    windowedTest([[], 3, { step: 2 }], []);
+    windowedTest([[], 3, { partial: true }], []);
+    windowedTest([[], 3, { step: 2, partial: true }], []);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] default option",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5], [
+      [1, 2, 3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3], [
+      [1, 2, 3],
+      [2, 3, 4],
+      [3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1], [
+      [1],
+      [2],
+      [3],
+      [4],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] step was set",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5, { step: 2 }], [
+      [1, 2, 3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3, { step: 2 }], [
+      [1, 2, 3],
+      [3, 4, 5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1, { step: 2 }], [
+      [1],
+      [3],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] partial was set",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5, { partial: true }], [
+      [1, 2, 3, 4, 5],
+      [2, 3, 4, 5],
+      [3, 4, 5],
+      [4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3, { partial: true }], [
+      [1, 2, 3],
+      [2, 3, 4],
+      [3, 4, 5],
+      [4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1, { partial: true }], [
+      [1],
+      [2],
+      [3],
+      [4],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] step and partial was set",
+  fn() {
+    windowedTest([[1, 2, 3, 4, 5], 5, { step: 2, partial: true }], [
+      [1, 2, 3, 4, 5],
+      [3, 4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 3, { step: 2, partial: true }], [
+      [1, 2, 3],
+      [3, 4, 5],
+      [5],
+    ]);
+    windowedTest([[1, 2, 3, 4, 5], 1, { step: 2, partial: true }], [
+      [1],
+      [3],
+      [5],
+    ]);
+  },
+});
+
+Deno.test({
+  name: "[collections/windowed] size or step was invalid",
+  fn() {
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], 0);
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], 3, { step: 0 });
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], NaN);
+      },
+      RangeError,
+      "Both size and step must be positive.",
+    );
+    assertThrows(
+      () => {
+        windowed([1, 2, 3, 4, 5], 3, { step: NaN });
```

Comment:
You could add tests for floats (like `0.5`, `1.2`), `Number.POSITIVE_INFINITY` and `Number.NEGATIVE_INFINITY`